README.md

```diff
@@ -1,85 +1,150 @@
-new CdkEksBlueprintStack(app, 'east-test-1', [new MetricsServerAddon, new ClusterAutoScaler, new ContainerInsightsAddOn], [], {
+new CdkEksBlueprintStack(app, 'east-test-1', addOns, [], {
   env: {
-      account: YOUR_ACCOUNT,
-      region: YOUR_TARGET_REGION,
+      account: <AWS_ACCOUNT_ID>,
+      region: <AWS_REGION>,
   },
 });
 ```
-Please create issues to request additional addons from the AWS portfolio. Otherwise, addons could be created outside of this repository, however a PR to the README maybe helpful to reference and advertise. 
+
+Deploy the stack 
+
+```
+cdk deploy
+```
+
+### Configuration
+
+Supports context variables (specify in cdk.json, cdk.context.json or pass with -c command line option):
+
+- `instanceType`: (defaulted to "t3.medium") Type of instance for the EKS cluster, must be a valid instance type like t3.medium
+- `vpc`: Specifies whether to use an existing VPC (if specified) or create a new one if not specified.
+- `minSize`: Min cluster size, must be positive integer greater than 0 (default 1).
+- `maxSize`: Max cluster size, must be greater than minSize.
+- `vpcSubnets`: List of VPC subnets for cluster provisioning (unsupported yet)
+
+## Solution Details
+
+### Goals
+
+The goal of this project is to provide a reference implementation of a Shared Services Platform built on top of EKS. At present the implementation provides the following functionality:
+
+  * Provision one or many EKS clusters across one or many regions.
+  * A modular approach to configuring the clusters with suite of add-ons or plugins that are needed to run workloads in a Kubernetes environment. 
+  * Add your own add-ons by implementing a `ClusterAddon` SPI (to be extended for lifecycle management). 
+  * Seamless onboarding of tenants/workloads onto specific clusters via CDK configuration and Gitops.
+
+### Supported Addons
+
+| AddOn             | Description                                                                       |
+|-------------------|-----------------------------------------------------------------------------------|
+| AppMesh           | Adds an AppMesh controller and CRDs (pending validation on the latest version of CDK) |
+| ArgoCD            | Adds an ArgoCD controller |
+| Calico            | Adds the Calico 1.7.1 CNI/Network policy engine |
+| CloudWatch        | Adds Container Insights support integrating monitoring with CloudWatch |
+| ClusterAutoscaler | Adds the standard cluster autoscaler ([Karpenter](https://github.com/awslabs/karpenter) is coming)|
+| MetricsServerAddon| Adds metrics server (pre-req for HPA and other monitoring tools)|
+| NginxAddon        | Adds NGINX ingress controller |
+
+### EKS Cluster Management 
+
+// Todo - Add
+
+### Configuring Addons 
+
+// Todo - Add
+
+### Creating an Addon
+
+// Todo - Add
+
+### Onboarding Tenants
+
+Examples are provided under `lib/teams`, however due to uniqueness of every team, clients are expected to supply implementation of the `TeamSetup` interface.
```

Comment:
I can add a typical base class for team setup that will do the following:

1. Create namespace based on the team name
2. Set quotas and policies (allow passing those as config values)
3. Optionally create a readonly user entry for troubleshooting and output the credentials in the stack

(this and other ideas like that can go into the backlog)